skllm/models/_base/classifier.py

```diff
@@ -211,24 +212,30 @@ def fit(
         self.classes_, self.probabilities_ = self._get_unique_targets(y)
         return self
 
-    def predict(self, X: Union[np.ndarray, pd.Series, List[str]]):
+    def predict(self, X: Union[np.ndarray, pd.Series, List[str]], num_workers: int = 1):
         """
         Predicts the class of each input.
 
         Parameters
         ----------
         X : Union[np.ndarray, pd.Series, List[str]]
             The input data to predict the class of.
+            
+        num_workers : int
+            number of workers to use for multithreaded prediction, default 1
 
         Returns
         -------
         np.ndarray
             The predicted classes as a numpy array.
         """
         X = _to_numpy(X)
-        predictions = []
-        for i in tqdm(range(len(X))):
-            predictions.append(self._predict_single(X[i]))
+        
+        if num_workers > 1:
+            raise DeprecationWarning("Passing num_workers to predict is temporary and will be removed in the future.")
```

Comment:
I don't think raising is preferred here as it interrupts the execution. I would suggest using `warnings.warn` instead.

Reply:
fixed